src/main/java/patching/Diff.java

```diff
@@ -313,4 +256,79 @@ else if (!other.insertion && !intermediate.insertion) {
         }
         return intermediateDiffs;
     }
+
+    private void transformType1(List<Diff> newIntermediateDiffs, Diff current, Diff other) {
+        newIntermediateDiffs.add(current);
+    }
+
+    private void transformType2(List<Diff> newIntermediateDiffs, Diff current, Diff other) {
+        int newStartLoc = current.startIndex + other.getLength();
+        Diff newDiff = new Diff(current.insertion, newStartLoc, current.changes);
+        newIntermediateDiffs.add(newDiff);
+    }
+
+    private void transformType3(List<Diff> newIntermediateDiffs, Diff current, Diff other) {
+        if ((other.startIndex + other.getLength()) > current.startIndex) {
+            int newStartLoc = current.startIndex - (current.startIndex - other.startIndex);
+            Diff newDiff = new Diff(current.insertion, newStartLoc, current.changes);
+            newIntermediateDiffs.add(newDiff);
+        } else {
+            int newStartLoc = current.startIndex - other.getLength();
+            Diff newDiff = new Diff(current.insertion, newStartLoc, current.changes);
+            newIntermediateDiffs.add(newDiff);
+        }
+    }
+
+    private void transformType4(List<Diff> newIntermediateDiffs, Diff current, Diff other) {
+        if ((other.startIndex + other.getLength()) <= current.startIndex) {
+            int newStartLoc = current.startIndex - other.getLength();
+            Diff newDiff = new Diff(current.insertion, newStartLoc, current.changes);
+            newIntermediateDiffs.add(newDiff);
+        } else if ((other.startIndex + other.getLength()) >= (current.startIndex
+                + current.getLength())) {
+            // Do nothing
```

Comment:
Yep. Again, this was just for clarity, and to match what we have in the OT effects table. I've added a little more info to explain what do nothing means.